__init__.py

```diff
@@ -55,89 +74,103 @@ def initialize(self):
-        """Show UI screens at a consistent interval."""
+            self.connect_to_wifi()
+
+    @intent_handler("test-gui.intent")
+    def test_gui(self, _):
+        """Show GUI screens at a consistent interval.
+
+        Testing wifi setup is difficult because the device must be in a certain state
+        for this skill to activate.  This method allows a developer to test changes to
+        the GUI without the difficulty of getting the device in a wifi setup state.
+        """
+        pages = [
+            "access_point_select", "follow_prompt", "network_select", "wifi_success"
+        ]
+        for page in pages:
+            self._show_page(page)
+            sleep(10)
+
+    def connect_to_wifi(self):
+        """Connect the device to a wifi network.
+
+        Mycroft devices need an internet connection for various purposes, such as
+        communicating with Selene.  If a device is already connected to the internet,
+        this setup logic should be skipped.
+        """
+        step_timeout = 8
+        self.schedule_repeating_event(
+            self._check_for_internet_connection,
+            when=None,
+            frequency=2,
+            name="InternetConnectCheck"
+        )
         steps = [
-            self.prompt_to_join_ap,
-            self.prompt_to_sign_in_to_ap,
-            self.prompt_to_select_network,
+            self._prompt_to_select_access_point,
+            self._prompt_to_select_wifi_network,
+            self._display_select_wifi_network,
         ]
         for step in steps:
-            if connected():
+            if self.connected_to_internet:
                 break
             step()
-            for sec in range(8):
-                if self.check_connection():
-                    return
-                else:
-                    sleep(1)
-            wait_while_speaking()
+            sleep_duration = 0
+            while not self.connected_to_internet and sleep_duration < step_timeout:
+                sleep(1)
+                sleep_duration += 1
 
+        while not self.connected_to_internet:
+            self.log.debug("No internet connection detected, waiting...")
+            sleep(2)
 
-        while True:
-            if self.check_connection():
-                return
-            else:
-                sleep(2)
-
+        self.cancel_scheduled_event("InternetConnectCheck")
+        self._report_setup_complete()
 
-    def prompt_to_join_ap(self, _=None):
+    def _prompt_to_select_access_point(self):
         """Prompt user to join temporary access point."""
-        self.speak_dialog("1_ap.created_speech")
-        self.gui["phone_image1"] = "1_phone_connect-to-ap.png"
-        self.gui["prompt1"] = "Connect to the \nWifi network"
-        self.gui["highlight1"] = "Mycroft"
-        self.gui.show_page("prompt1.qml", override_idle=True)
+        self._show_page("access_point_select")
+        self.speak_dialog("access_point_created", wait=True)
```

Comment:
Ah just realised it's an underscore vs dash thing. The filename is `access-point-created`

Reply:
fixed